app/src/test/java/com/asfoundation/wallet/promotions/PromotionsInteractorTest.kt

```diff
@@ -0,0 +1,343 @@
+        .thenReturn(Single.just(VoucherListModel(
+            listOf(Voucher("test", "Trivial Drive Sample", "icon", true)))))
+
+    val linkItem = GamificationLinkItem("id3", "description", null, TEST_START_DATE, TEST_END_DATE)
+    val gamificationItem =
+        GamificationItem(GAMIFICATION_ID, null, 1, 0, "title", BigDecimal(9), 20.0, 25.0,
+            listOf(linkItem).toMutableList())
+    val voucherItem = VoucherItem(VOUCHER_ID, "test", "Trivial Drive Sample", "icon", true, 25.0)
+    val progressItem =
+        ProgressItem("id1", "description", null, TEST_START_DATE,
+            TEST_END_DATE, BigDecimal(50), BigDecimal(100), "any")
+    val defaultItem =
+        DefaultItem("id2", "description", null, TEST_START_DATE, TEST_END_DATE,
+            null)
+
+    val promotionsModel = interactor.retrievePromotions()
+        .blockingGet()
+    val testPromotionsModel = PromotionsModel(listOf(gamificationItem), listOf(voucherItem),
+        listOf(progressItem, defaultItem, linkItem),
+        com.asfoundation.wallet.promotions.WalletOrigin.APTOIDE, null)
+
+    Assert.assertEquals(promotionsModel.promotions.size, testPromotionsModel.promotions.size)
+    promotionsModel.promotions.forEachIndexed { index, promotion ->
+      if (promotion is GamificationItem) {
+        compareGamificationItem(promotion, testPromotionsModel.promotions[index])
+      }
+    }
+    promotionsModel.vouchers.forEachIndexed { index, voucher ->
+      compareVoucherItem(voucher, testPromotionsModel.vouchers[index])
+    }
+    promotionsModel.perks.forEachIndexed { index, perkPromotion ->
+      if (perkPromotion is DefaultItem) {
+        compareDefaultItem(perkPromotion, testPromotionsModel.perks[index] as DefaultItem)
+      } else if (perkPromotion is ProgressItem) {
+        compareProgressItem(perkPromotion, testPromotionsModel.perks[index] as ProgressItem)
+      }
+    }
+    Assert.assertEquals(promotionsModel.walletOrigin, testPromotionsModel.walletOrigin)
+    Assert.assertEquals(promotionsModel.error, testPromotionsModel.error)
+  }
+
+  @Test
+  fun hasAnyPromotionUpdateNegativeTest() {
+    Mockito.`when`(
+        referralInteractor.hasReferralUpdate(TEST_WALLET_ADDRESS, null, ReferralsScreen.PROMOTIONS))
+        .thenReturn(Single.just(false))
+    Mockito.`when`(gamificationInteractor.hasNewLevel(TEST_WALLET_ADDRESS, gamificationResponse,
+        GamificationContext.SCREEN_PROMOTIONS))
+        .thenReturn(Single.just(false))
+    Mockito.`when`(promotionsRepository.getSeenGenericPromotion(Mockito.anyString(),
+        Mockito.anyString()))
+        .thenReturn(true)
+    Mockito.`when`(userLocalData.getSeenWalletOrigin(TEST_WALLET_ADDRESS))
+        .thenReturn(testWalletOrigin.name)
+    val hasPromotionUpdate = interactor.hasAnyPromotionUpdate(PromotionUpdateScreen.PROMOTIONS)
+        .blockingGet()
+    Assert.assertFalse(hasPromotionUpdate)
+  }
+
+  @Test
+  fun hasAnyPromotionUpdatePositiveTest() {
+    Mockito.`when`(
```

Comment:
This technically is not testing every possible combination of values where they end up being a positive case but don't think it's important and I don't think it would be very easy to test (can we even for loop it somehow??)